### PR TITLE
Improve nested_dict serialization test

### PR DESCRIPTION
**Pull request by hussein-awala**
related: #30819 

This PR replace `pandas.Dataframe` test data by a simple test class, and export the list of data to make the test parametrized.

### CODE REVIEW

**Comment by bolkedebruin** [comment]
Can pandas be removed from the requirement for this test?

**Comment by potiuk** [comment]
> Can pandas be removed from the requirement for this test?

Or rather have another test with no-pandas ? I think it's cool to have both

**Comment by bolkedebruin** [comment]
I'd prefer the simplest test case and not bring in a heavy requirement like pandas for this. It makes it harder to read and understand what is going on

**Comment by potiuk** [comment]
> I'd prefer the simplest test case and not bring in a heavy requirement like pandas for this. It makes it harder to read and understand what is going on

we can separate the pandas test to another module/test-case -> we already have (And use) pandas in our test suite and CI image so I see no harm in having both in the way that it does not require you to have pandas to run the "basic" test suite

File: tests/serialization/test_serde.py [hunk 1]
```diff
@@ -309,3 +309,31 @@ def test_stringify(self):
...
+            assert len(actual_obj) == len(expected_obj)
+            for key in actual_obj:
+                if isinstance(actual_obj[key], dict):
+                    return _compare(actual_obj[key], expected_obj[key])
+                if isinstance(actual_obj[key], pandas.DataFrame):
+                    assert actual_obj[key].equals(expected_obj[key])
+                else:
+                    assert actual_obj[key] == expected_obj[key]
+
+        data = (
```
Comment (bolkedebruin) [inline tests/serialization/test_serde.py]: why not use `@pytest.mark.parametrize` here, makes it consistent with other test cases.

**Comment by bolkedebruin** [comment]
> > I'd prefer the simplest test case and not bring in a heavy requirement like pandas for this. It makes it harder to read and understand what is going on
> 
> we can separate the pandas test to another module/test-case -> we already have (And use) pandas in our test suite and CI image so I see no harm in having both in the way that it does not require you to have pandas to run the "basic" test suite

We can, but I do think it is redundant and I prefer atomic tests particularly here.

**Comment by hussein-awala** [comment]
@bolkedebruin It seems like the serialize/deserialize methods didn't have any issue, but it was just the encoders and decoders which call them, so I'll keep the test in the module `test_json` but I replaced `pandas.Dataframe` by the test class `Z` (I confirm that the problem was with the nested classes including the class `Z`), and I used parametrized tests

**Comment by potiuk** [comment]
Yep. If class Z also shows the problem, I am all in for removing pandas.

**Review by potiuk — APPROVED** [review]

**Comment by bolkedebruin** [comment]
Can you add a simple test in `test_serde.py` that tries to deserialize an existing/non-serialized object? And add a note to where you made the fix that or to that test as to why it is there (a decoder can travel both ways)? 

Furthermore, I am not entirely happy with factoring Z out and putting it at the module level. Why not keep it in test_serde and reference it from the test_json? Or have a separate implementation in test_json as Z itself isnt really important.

**Review by bolkedebruin — APPROVED** [review]

**Pull request merged**